lib/session/messageSession.ts

```diff
@@ -0,0 +1,887 @@
+          // Nothing much to do if user's message handler throws. Let us try abandoning the message.
+          if (this.receiveMode === ReceiveMode.peekLock) {
+            try {
+              log.error("[%s] Abandoning the message with id '%s' on the receiver '%s' since " +
+                "an error occured: %O.", connectionId, bMessage.messageId, this.name, error);
+              await bMessage.abandon();
+            } catch (abandonError) {
+              const translatedError = translate(abandonError);
+              log.error("[%s] An error occurred while abandoning the message with id '%s' on the " +
+                "receiver '%s': %O.", connectionId, bMessage.messageId, this.name, translatedError);
+              this._notifyError(translatedError);
+            }
+          }
+          return;
+        }
+
+        // If we've made it this far, then user's message handler completed fine. Let us try
+        // completing the message.
+        if (this.autoComplete && this.receiveMode === ReceiveMode.peekLock) {
+          try {
+            log.messageSession("[%s] Auto completing the message with id '%s' on " +
+              "the receiver '%s'.", connectionId, bMessage.messageId, this.name);
+            await bMessage.complete();
+          } catch (completeError) {
+            const translatedError = translate(completeError);
+            log.error("[%s] An error occurred while completing the message with id '%s' on the " +
+              "receiver '%s': %O.", connectionId, bMessage.messageId, this.name, translatedError);
+            this._notifyError(translatedError);
+          }
+        }
+      };
+      // setting the "message" event listener.
+      this._receiver.on(ReceiverEvents.message, onSessionMessage);
+      // adding credit
+      this._receiver!.setCreditWindow(this.maxConcurrentCallsPerSession);
+      this._receiver!.addCredit(this.maxConcurrentCallsPerSession);
+    } else {
+      this._isReceivingMessages = false;
+      const msg = `MessageSession with sessionId '${this.sessionId}' and name '${this.name}' ` +
+        `has either not been created or is not open.`;
+      log.error("[%s] %s", this._context.namespace.connectionId, msg);
+      this._notifyError(new Error(msg));
+    }
+  }
+
+  /**
+   * Receive a batch of Message objects from a ServiceBus Queue/Topic for a given count and
+   * a given max wait time in seconds, whichever happens first. This method can be used directly
+   * after creating the receiver object and **MUST NOT** be used along with the `start()` method.
+   * @param maxMessageCount The maximum message count. Must be a value greater than 0.
+   * @param maxWaitTimeInSeconds The maximum wait time in seconds for which the Receiver
+   * should wait to receiver the said amount of messages. If not provided, it defaults to 60 seconds.
+   */
+  receiveBatch(maxMessageCount: number, maxWaitTimeInSeconds?: number): Promise<ServiceBusMessage[]> {
```

Comment:
How about `maxMessageWaitTimeoutInSeconds` just like you added in the `batchingReceiver`?